### PR TITLE
Add generic function to intersect slices

### PR DESCRIPTION
**Pull request by tpantelis**

### CODE REVIEW

File: pkg/slices/slices_test.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+				actual := slices.Intersect(s1, s2, key)
+				Expect(sets.New(actual...).Equal(sets.New(exp...))).To(BeTrue(), "Expected: %s. Actual: %s", exp, actual)
+			}
+
+			testIntersect([]string{"1", "2"}, []string{"1", "2"}, "1", "2")
+			testIntersect([]string{"1", "2"}, []string{"2", "1"}, "1", "2")
+			testIntersect([]string{"1", "2"}, []string{"1"}, "1")
+			testIntersect([]string{"3"}, []string{"1", "2", "3"}, "3")
+			testIntersect([]string{"1", "2", "3", "4"}, []string{"2", "5", "6", "3"}, "2", "3")
+		})
```
Comment (skitt) [inline pkg/slices/slices_test.go:42]: This should also have a test returning an empty set.

**Review by skitt — CHANGES_REQUESTED** [review]

File: pkg/slices/slices_test.go (hunk 1 above)
Comment (tpantelis) [reply pkg/slices/slices_test.go:42]: The next test case on L44 does that.

Comment (skitt) [reply pkg/slices/slices_test.go:42]: Let me clarify:

```suggestion
			testIntersect([]string{"1", "2", "3", "4"}, []string{"5", "6", "7", "8"})
		})
```

Comment (tpantelis) [reply pkg/slices/slices_test.go:42]: right  - thanks

**Review by skitt — APPROVED** [review]

**Comment by tpantelis** [comment]
@aswinsuryan

**Review by aswinsuryan — APPROVED** [review]
